assemblyline_v4_service/updater/updater.py

```diff
@@ -378,14 +378,38 @@ def do_source_update(self, service: Service) -> None:
                     source_obj = sources[source_name]
                     old_update_time = self.get_source_update_time()
 
-                    self.push_status("UPDATING", "Starting..")
-                    source = source_obj.as_primitives()
-                    uri: str = source['uri']
-                    default_classification = source.get('default_classification', classification.UNRESTRICTED)
-                    # Enable syncing if the source specifies it
-                    self.client.sync = source.get('sync', False)
-
+                    # Are we ignoring the cache for this source?
+                    if source_obj.ignore_cache:
+                        old_update_time = 0
                     try:
+
+                        source = source_obj.as_primitives()
+                        uri: str = source_obj.uri
+
+                        # If source is not currently enabled/active, skip..
+                        if not source_obj.enabled:
+                            raise SkipSource
+
+                        # Is it time for this source to run?
+                        elapsed_time = time.time() - old_update_time
+                        update_interval = source.get('update_interval') or service.update_config.update_interval_seconds
+                        if elapsed_time < update_interval:
+                            # Too early to run the update for this particular source, skip for now
+                            raise SkipSource
+
+
+                        self.push_status("UPDATING", "Starting..")
+                        fetch_method = source.get('fetch_method', 'GET')
+                        default_classification = source.get('default_classification', classification.UNRESTRICTED)
+
+                        # Configure the client as necessary
+
+                        # Enable syncing if the source specifies it
+                        self.client.sync = source.get('sync', False)
+                        # Override classfication of signatures if specified
+                        self.client.classification_override = default_classification \
+                            if source.get('override_classification', False) else None
```

Comment:
If we are to use two lines, I think this formatting makes it easier to read, and does not call the setter function when not needed.
```suggestion
                        if source.get('override_classification', False):
                            self.client.classification_override = default_classification
```

Reply:
```suggestion
                        # Reset client back to original classification state between updates
                        self.client.classification_override = None
                        if source.get('override_classification', False):
                            self.client.classification_override = default_classification
```